### PR TITLE
coding guidelines: comply with MISRA C:2012 Rule 8.3

### PR DESCRIPTION
**Pull request by Abramo-Bagnara**
In particular:

- use always the same parameter names in every redeclarations

Signed-off-by: Abramo Bagnara <abramo.bagnara@bugseng.com>

### CODE REVIEW

File: kernel/userspace.c [hunk 1]
```diff
@@ -901,19 +901,19 @@ SYS_INIT(app_shmem_bss_zero, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
...
 	arch_syscall_oops(ssf);
 	CODE_UNREACHABLE; /* LCOV_EXCL_LINE */
 }
 
-static uintptr_t handler_no_syscall(uintptr_t arg1, uintptr_t arg2,
-				    uintptr_t arg3, uintptr_t arg4,
-				    uintptr_t arg5, uintptr_t arg6, void *ssf)
+static uintptr_t handler_no_syscall(uintptr_t arg0, uintptr_t arg1,
+				    uintptr_t arg2, uintptr_t arg3,
+				    uintptr_t arg4, uintptr_t arg5, void *ssf)
```
Comment (peter-mitsis) [inline kernel/userspace.c:916]: I am guessing that the intent behind this change is to make the numbering of the arguments more consistent with that of  the other sycalls. However, like handler_bad_syscall(), I think scripts/gen_syscalls.py should be updated if moving ahead with this change.

File: lib/os/cbprintf_complete.c [hunk 2]
```diff
@@ -1372,7 +1372,7 @@ static int outs(cbprintf_cb out,
 	return count;
 }
 
-int cbvprintf(cbprintf_cb out, void *ctx, const char *fp, va_list ap)
+int cbvprintf(cbprintf_cb out, void *ctx, const char *format, va_list ap)
```
Comment (peter-mitsis) [inline lib/os/cbprintf_complete.c:1375]: This routine has two definitions. One here, and another in lib/os/cbprintf_nano.c. The one in cbprintf_nano.c should be updated as well.

File: kernel/userspace.c [hunk 3]
```diff
@@ -901,19 +901,19 @@ SYS_INIT(app_shmem_bss_zero, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
  * Default handlers if otherwise unimplemented
  */
 
-static uintptr_t handler_bad_syscall(uintptr_t bad_id, uintptr_t arg2,
-				     uintptr_t arg3, uintptr_t arg4,
-				     uintptr_t arg5, uintptr_t arg6,
+static uintptr_t handler_bad_syscall(uintptr_t bad_id, uintptr_t arg1,
+				     uintptr_t arg2, uintptr_t arg3,
+				     uintptr_t arg4, uintptr_t arg5,
```
[the remark this answers is not in the document]
Comment (dcpleung) [reply kernel/userspace.c:906]: This is going to cause confusion as `arch_syscall_invoke*()` start with `arg1`.

File: kernel/userspace.c (hunk 1 above)
Comment (dcpleung) [reply kernel/userspace.c:916]: This is going to cause confusion as `arch_syscall_invoke*()` start with `arg1`.

File: include/sys/arch_interface.h [hunk 4]
```diff
@@ -290,15 +290,15 @@ int arch_irq_is_enabled(unsigned int irq);
...
+ * @param arg ISR parameter
  * @param flags Arch-specific IRQ configuration flag
  *
  * @return The vector assigned to this interrupt
  */
 int arch_irq_connect_dynamic(unsigned int irq, unsigned int priority,
-			     void (*routine)(const void *parameter),
-			     const void *parameter, uint32_t flags);
+			     void (*func)(const void *arg),
+			     const void *arg, uint32_t flags);
```
[the remark this answers is not in the document]
Comment (Abramo-Bagnara) [reply include/sys/arch_interface.h]: Done.

**Review by peter-mitsis — APPROVED** [review]

**Pull request merged**